.github/workflows/workflow.yml

```diff
@@ -0,0 +1,29 @@
+name: Test
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+     - uses: actions/checkout@v2
+
+     - name: Set up Python 3.9
+       uses: actions/setup-python@v2
+       with:
+         python-version: 3.9
+
+     - name: Install dependencies
+       shell: bash
+       run: |
+         sudo apt-get install libsasl2-dev
+         python -m pip install --upgrade pip
+         python -m pip install -r test-requirements.txt
+
+     - name: Run pytest
+       shell: bash
+       run: DBT_PROFILES_DIR=$PWD pytest tests/
```

Comment:
I see - this tries to run `tests/functional/test_utils.py` as well (added in https://github.com/dbt-labs/spark-utils/pull/25), which has other requirements (specified in `dev-requirements.txt`).

Options:
1. **Unify:** Add `python -m pip install -r dev-requirements.txt` into the "Install dependencies" step above—or better yet, unify `dev-requirements` + `test-requirements` into a single file—and try running the tests with `--profile session`. I imagine that a few may fail (as in `dbt-spark`), but the rest should succeed, and we can mark the failing ones to skip on the "session" profile.
2. **Separate:** Treat these as two totally independent testing frameworks, which is really what they are. Create `tests/functional` for the tests using the `dbt-core` functional-testing framework, and `tests/unit` for the tests using the `pytest-dbt-core` unit-testing framework. Have clearly separate READMEs and files used by each: separate requirements, separate profile setup, all that jazz.

What do you think? I lean toward option 2, "separate" — and then `tests/unit` could be a clear, self-contained, and real-world demonstration of the potential of `pytest-dbt-core`

Reply:
+1 to option 2, "separate" (especially different folders `tests/unit` vs. `tests/functional`).

Separate folders won't undermine the functionality at all, and it will make the delineations between different frameworks more clear.

Side note: I don't see a problem with them sharing a single `dev-requirements.txt` though.

Reply:
I moved the `test_macros.py` in a `unit` subfolder. And, I merged the `test-requirements.txt` into the `dev-requirements.txt`. I think it is confusing to have a `test-requirements.txt` that does not contain all dependencies for the functional tests. 

I had to move the `conftest.py` into the `functional` subdirectory because it was interfering with the unit tests.